applications/SolversApplication/linear_system/system_builders/system_builder_and_solver.hpp

```diff
@@ -0,0 +1,545 @@
+//
+//   Project Name:        KratosSolversApplication $
+//   Created by:          $Author:     JMCarbonell $
+//   Last modified by:    $Co-Author:              $
+//   Date:                $Date:      January 2019 $
+//   Revision:            $Revision:           0.0 $
+//
+//
+
+#if !defined(KRATOS_SYSTEM_BUILDER_AND_SOLVER_H_INCLUDED)
+#define KRATOS_SYSTEM_BUILDER_AND_SOLVER_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_solvers/solution_schemes/solution_scheme.hpp"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+
+/** Short class definition.
+Current class provides an implementation for standard builder and solving operations.
+
+the RHS is constituted by the unbalanced loads (residual)
+
+Degrees of freedom are reordered putting the restrained degrees of freedom at
+the end of the system ordered in reverse order with respect to the DofSet.
+
+Imposition of the dirichlet conditions is naturally dealt with as the residual already contains this information.
+ */
+
+
+/** @brief System Buider and Solver base class
+ *  @details This is the base class for the building and solving the system
+ */
+template<class TSparseSpace,
+         class TDenseSpace, // = DenseSpace<double>,
+         class TLinearSolver //= LinearSolver<TSparseSpace,TDenseSpace>
+         >
+class SystemBuilderAndSolver : public Flags
```

Comment:
Why creating a new class thst is exactly the same that the BuilderAndSolver class. If anything relevant is changed in the elements or conditions is changed all this new classes will need to de recoded again